src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -219,7 +219,7 @@ private static bool TryGetByteArrayFromFileStream(Stream stream, out byte[] buff
             // than the buffer size. The default buffer size is 4KB, so this will incur a 4KB
             // allocation for any files less than 4KB. That's why, for example, the command
             // line compiler actually specifies a very small buffer size.
-            return stream.Read(buffer, 0, length) == length;
+            return stream.TryReadAll(buffer, 0, length) == length;
```

Comment:
We will be doing additional reads only if the stream doesn't give us what we want in one read, which almost never happens on desktop. Besides, we want to read all the data, so what is the alternative?